### ISSUE TITLE
Use of OpenApi annotations leads to slightly imprecise descriptions in the Admin REST API

### ISSUE THREAD

**Issue by christian-2** [keycloak/keycloak#21495]
### Before reporting an issue

- [X] I have searched existing issues
- [X] I have reproduced the issue with the [latest nightly release](https://github.com/keycloak/keycloak/releases/tag/nightly)

### Area

admin/api

### Describe the bug

The documentation for the [Keycloak Admin REST API](https://www.keycloak.org/docs-api/21.1.2/rest-api/index.html) describes the two methods `GET /{realm}/roles` and `GET /{realm}/clients/{id}/roles` alike as "Get all roles for the realm or client", whereas it would seem that the first method "Gets all roles for the realm" and the second one "Gets all roles for the client". The same holds for other pairs of methods as well: e.g. `POST /{realm}/roles` and `POST /{realm}/clients/{id}/roles` are both described as "Create a new role for the realm or client".

I assume this is a bug in the sense that the documentation is slightly off (too general) because both methods in the pair are implemented by the same piece of code (in [`RoleContainerResource`](https://github.com/keycloak/keycloak/blob/main/services/src/main/java/org/keycloak/services/resources/admin/RoleContainerResource.java)), which injects the same description by means of a single OpenApi annotation (`@Operation`).

I find the observed discrepancies a bit confusing, and I therefore wonder whether OpenApi (as used by Keycloak) perhaps offers betters support for such corner cases. 

### Version

21.1.2

### Expected behavior

descriptions should be specific and hence distinct (see above)

### Actual behavior

descriptions are the same and hence somewhat imprecise (see above)

### How to Reproduce?

look it up in the current documentation (see above)

### Anything else?

a minor bug for sure, but presumably still a bug

**Comment by christian-2** [comment]
As a workaround, one could change summary texts in [`RoleContainerResource.java`](https://github.com/keycloak/keycloak/blob/main/services/src/main/java/org/keycloak/services/resources/admin/RoleContainerResource.java) (this example, plus analog treatment for instances concerning `POST`, etc.):

```
@Operation( summary = "Get all roles for the realm (GET /{realm}/roles) or client (GET /{realm}/clients/{id}/roles)")
```

But clearly, it would be better if OpenApi had specific support for such polymorphism (which I think it does).

**Comment by christian-2** [comment]
Somewhat related: #8983

**Comment by mposolda** [comment]
I am changing to enhancement and adding to Backlog as I am not sure if Keycloak team has time to look at this in the near future. Anyone is welcome to investigate and eventually contribute some good solution for this.

TBH The approach from the comment https://github.com/keycloak/keycloak/issues/21495#issuecomment-1628216601 seems to me like another workaround, which rather makes documentation more complicated, but maybe there is some possibility...

**Comment by christian-2** [comment]
@mposolda yes, I agree that this fits into backlog and that my comment should not be taken as the solution. Maybe you want to also label this as `help wanted`.
